tests/tests/bm25_search.rs

```diff
@@ -379,6 +379,15 @@ fn snippet(mut conn: PgConnection) {
     assert_eq!(row.0, 3);
     assert_eq!(row.1, "Sleek running <h1>shoes</h1>");
     assert_relative_eq!(row.2, 2.484906, epsilon = 1e-6);
+
+    let row: (i32, String, f32) = "
+    SELECT id, paradedb.snippet(description, '<h1>', '</h1>', max_num_chars=>'2'), paradedb.score(id)
+    FROM paradedb.bm25_search WHERE bm25_search @@@ 'description:shoes' ORDER BY id"
+    .fetch_one(&mut conn);
+
+    assert_eq!(row.0, 3);
+    assert_eq!(row.1, "<h1>shoes</h1>");
+    assert_relative_eq!(row.2, 2.484906, epsilon = 1e-6);
```

Comment:
You're passing `'2'` here, a string, to `max_num_chars`. It should be an integer.

I'm not sure this test is showing that the parameter is working correctly? It's still returning the full match for `shoes` as  `"<h1>shoes</h1>"`.

Reply:
I did a test using `tantivy`, the result should be correct
```rust
#[cfg(test)]
mod tests {

    use tantivy::{
        doc,
        query::QueryParser,
        schema::{Schema, TEXT},
        snippet::SnippetGenerator,
        Index,
    };

    #[test]
    fn test_snippet_from_doc() {
        let mut schema_builder = Schema::builder();
        let text_field = schema_builder.add_text_field("text", TEXT);
        let schema = schema_builder.build();
        let index = Index::create_in_ram(schema);
        let mut index_writer = index.writer_with_num_threads(1, 20_000_000).unwrap();
        let doc = doc!(text_field => r#"Comme je descendais des Fleuves impassibles,
  Je ne me sentis plus guidé par les haleurs :
 Des Peaux-Rouges criards les avaient pris pour cibles,
 Les ayant cloués nus aux poteaux de couleurs.
 J'étais insoucieux de tous les équipages,
 Porteur de blés flamands ou de cotons anglais.
 Quand avec mes haleurs ont fini ces tapages,
 Les Fleuves m'ont laissé descendre où je voulais.
     "#);
        index_writer.add_document(doc.clone()).unwrap();
        index_writer.commit().unwrap();
        let query_parser = QueryParser::for_index(&index, vec![text_field]);
        let query = query_parser.parse_query("haleurs flamands").unwrap();
        let reader = index.reader().unwrap();
        let searcher = reader.searcher();
        let mut snippet_generator =
            SnippetGenerator::create(&searcher, &*query, text_field).unwrap();
        snippet_generator.set_max_num_chars(100);
        let snippet = snippet_generator.snippet_from_doc(&doc);
        let snippet_html: String = snippet.to_html();
        assert_eq!(snippet_html, "Comme je descendais des Fleuves impassibles,\n  Je ne me sentis plus guidé par les <b>haleurs</b> :\n Des");

        snippet_generator.set_max_num_chars(2);
        let snippet = snippet_generator.snippet_from_doc(&doc);
        let snippet_html: String = snippet.to_html();
        assert_eq!(snippet_html, "<b>haleurs</b>");
    }
}

```